SheCodesAus/2024_a_dynamic_array_frontend
# Rachel/update password

## Describe your changes

Password update - React

- created the post-update-password API
- created the UpdatePassword.jsx form under components>forms
- created the UpdatePassword.css under components>forms (although no styling applied as I relied on existing styling by using existing className on form elements. 
- created the UpdatePasswordPage.jsx under pages and inserted this into main.jsx
- I have tested this extensively. User must be logged in, token is passed into the PUT API request to update-password, the view then checks token and authorizes user before performing password checks in the serializer. The new password is saved and is saved correctly in hashed format. 


Code fixes of existing code base:
- fixed bug in LoginPage (changed "class" to "className" in one of the divs)
- Removed <LoginForm /> after the LoginPage() LoginPage.jsx that shouldn't be there.


## Type of change


## What kind of feedback do you need?

Review and test. I'll work on getting the backend PR merged so that this can be easily tested. 

How to test:
- ensure you are working with the most up to date backend DEV branch
- running two VSCode instances, npm run server and python manage.py runserver, in React API files, ensure that the local url link is employed (instructions are nested in the code here). Be sure to revert back to remote url before pushing any code yourself!
- login as a user
- navigate to /update-password to have a go at changing your password!

## Checklist:

- [ ] I have recommended where to start and how to proceed with the review
